sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyType.cs

```diff
@@ -3,60 +3,60 @@
 // license information.
 
 using System;
+using System.Collections.Generic;
 
 namespace Azure.Security.KeyVault.Keys
 {
     /// <summary>
     /// Supported JsonWebKey key types (kty)
     /// </summary>
-    public enum KeyType : uint
+    public struct KeyType
     {
-        EllipticCurve = 0x0001,
-        EllipticCurveHsm = 0x0002,
-        Rsa = 0x0004,
-        RsaHsm = 0x0008,
-        Octet = 0x0010,
-        Other = 0x0020,
-    }
+        /// <summary>
+        /// Supported JsonWebKey key types (kty) as enum
+        /// </summary>
+        public enum KeyTypeValues : uint
+        {
+            EllipticCurve = 0x0001,
+            EllipticCurveHsm = 0x0002,
+            Rsa = 0x0004,
+            RsaHsm = 0x0008,
+            Octet = 0x0010
+            // When a new entry is added, make sure to expose it as a field below.
+        }
 
-    public static class KeyTypeExtensions
-    {
-        public static KeyType ParseFromString(string value)
+        public static readonly KeyType Rsa = new KeyType("RSA");
+        public static readonly KeyType RsaHsm = new KeyType("RSA-HSM");
+        public static readonly KeyType EllipticCurve = new KeyType("EC");
+        public static readonly KeyType EllipticCurveHsm = new KeyType("EC-HSM");
+        public static readonly KeyType Octet = new KeyType("Octet");
+
+        public string StringValue { get; private set; }
```

Comment:
I think we should add a field of the enum type, add Other enum value, and override ToString to let people access optional string value field.